src/clj/fluree/db/reasoner/owl_datalog.cljc

```diff
@@ -434,14 +401,19 @@
                   acc)))
 
           union-of
-          (-> acc
-              (conj {"@id"    binding-var
-                     property "?_some-val-rel"})
-              (conj (reduce (fn [acc i]
-                              (conj acc {"@id"   "?_some-val-rel"
-                                         "@type" {"@id" i}}))
-                            ["union"]
-                            (-> union-of first (get const/iri-owl:unionOf) get-all-ids))))
+          (let [union-classes   (-> union-of
+                                    first ;; always sequential, but only can be one value so take first
+                                    (get-named-ids const/iri-owl:unionOf))
+                with-property-q {"@id"    binding-var
+                                 property "?_some-val-rel"}
+                of-classes-q    (reduce (fn [acc class]
+                                          (conj acc {"@id"   "?_some-val-rel"
+                                                     "@type" {"@id" class}}))
+                                        ["union"]
+                                        union-classes)]
+            (-> acc
```

Comment:
`conj` takes multiple arguments, so you can rewrite this as:

```clojure
(conj acc with-property-q of-classes-q)
```
instead of using `->`.

Reply:
Addressed in 70b957ff934dc64cc981bb38bfcfe6c943bf4dbc